drm/ResourceManager.cpp

```diff
@@ -71,6 +72,40 @@ void ResourceManager::Init() {
       if (stat(path.str().c_str(), &buf) != 0)
         break;
 
+      node_num++;
+    }
+
+    // only have one card, is BM/GVT-d/Virtio
+    if (node_num == 1) {
+      std::ostringstream path;
+      path << path_pattern << 0;
+      auto dev = DrmDevice::CreateInstance(path.str(), this);
+      if (dev) {
+        drms_.emplace_back(std::move(dev));
+      }
+    }
+    // is SR-IOV or iGPU + dGPU
+    if (node_num == 2) {
+      std::ostringstream path;
+      path << path_pattern << 1;
+      auto dev = DrmDevice::CreateInstance(path.str(), this);
+      if (dev->GetName() == "i915") {
+        std::ostringstream path;
+        path << path_pattern << 0;
+        auto dev = DrmDevice::CreateInstance(path.str(), this);
+        if (dev) {
+          drms_.emplace_back(std::move(dev));
+        }
+      } else {
+        if (dev) {
+          drms_.emplace_back(std::move(dev));
+        }
+      }
+    }
+    // is SRI-IOV + dGPU
+    if (node_num == 3) {
```

Comment:
suggest add more comments, also there is typo:
is SR-IOV + dGPU case, Celadon only use card2(virtio-gpu) for display